source/server-discovery-and-monitoring/tests/integration/auth-network-error.yml

```diff
@@ -0,0 +1,81 @@
+# Test SDAM error handling.
+runOn:
+    # failCommand appName requirements
+    - minServerVersion: "4.4"
+      requireAuth: true
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "auth-network-error"
+
+data: &data
+  - {_id: 1}
+  - {_id: 2}
+
+tests:
+  - description: Reset server and pool after network error during authentication
```

Comment:
a similar one for network timeouts (both auth and non-auth) could be nice to cover the last bit of the branch for handling pre-handshake errors.

Reply:
I added two more tests, one for network timeouts during auth and one for network timeouts post-auth.